api/v1_update_coin_test.go

```diff
@@ -0,0 +1,381 @@
+	assert.Equal(t, 200, status)
+	jsonAssert(t, body, map[string]any{
+		"data.mint":        "bearR26zyyB3fNQm5wWv1ZfN8MPQDUMwaAuoG79b1Yj",
+		"data.ticker":      "$BEAR",
+		"data.name":        "BEAR",
+		"data.description": "Updated description for the bear token",
+		"data.twitter":     "https://twitter.com/bear_token",
+		"data.instagram":   "https://instagram.com/bear_token",
+		"data.tiktok":      "https://tiktok.com/@bear_token",
+		"data.website":     "https://bear-token.com",
+	})
+}
+
+func TestV1UpdateCoin_CoinNotFound(t *testing.T) {
+	app := emptyTestApp(t)
+	database.Seed(app.pool.Replicas[0], database.FixtureMap{
+		"users": {
+			{
+				"user_id":     1,
+				"wallet":      "0x7d273271690538cf855e5b3002a0dd8c154bb060",
+				"is_verified": true,
+			},
+		},
+	})
+
+	requestBody := UpdateCoinBody{
+		Description: "Updated description",
+		Twitter:     "https://twitter.com/test",
+		Instagram:   "https://instagram.com/test",
+		Tiktok:      "https://tiktok.com/@test",
+		Website:     "https://test.com",
+	}
+	requestBodyBytes, err := json.Marshal(requestBody)
+	assert.NoError(t, err)
+
+	status, body := testPostWithWallet(t, app, "/v1/coins/nonexistentMint?user_id="+trashid.MustEncodeHashID(1), "0x7d273271690538cf855e5b3002a0dd8c154bb060", requestBodyBytes, map[string]string{
+		"Content-Type": "application/json",
+	})
+
+	assert.Equal(t, 404, status)
+	jsonAssert(t, body, map[string]any{
+		"error": "Coin not found",
+	})
+}
+
+func TestV1UpdateCoin_Unauthorized(t *testing.T) {
+	app := emptyTestApp(t)
+	database.Seed(app.pool.Replicas[0], database.FixtureMap{
+		"users": {
+			{
+				"user_id":     1,
+				"wallet":      "0x7d273271690538cf855e5b3002a0dd8c154bb060",
+				"is_verified": true,
+			},
+			{
+				"user_id":     2,
+				"wallet":      "0xc3d1d41e6872ffbd15c473d14fc3a9250be5b5e0",
+				"is_verified": true,
+			},
+		},
+		"artist_coins": {
+			{
+				"mint":        "bearR26zyyB3fNQm5wWv1ZfN8MPQDUMwaAuoG79b1Yj",
+				"ticker":      "$BEAR",
+				"user_id":     1, // Owned by user 1
+				"decimals":    9,
+				"name":        "BEAR",
+				"description": "Original description",
+			},
+		},
+	})
+
+	requestBody := UpdateCoinBody{
+		Description: "Updated description",
+		Twitter:     "https://twitter.com/test",
+		Instagram:   "https://instagram.com/test",
+		Tiktok:      "https://tiktok.com/@test",
+		Website:     "https://test.com",
+	}
+	requestBodyBytes, err := json.Marshal(requestBody)
+	assert.NoError(t, err)
+
+	// Try to update with user 2 (who doesn't own the coin)
+	status, body := testPostWithWallet(t, app, "/v1/coins/bearR26zyyB3fNQm5wWv1ZfN8MPQDUMwaAuoG79b1Yj?user_id="+trashid.MustEncodeHashID(2), "0xc3d1d41e6872ffbd15c473d14fc3a9250be5b5e0", requestBodyBytes, map[string]string{
+		"Content-Type": "application/json",
+	})
+
+	assert.Equal(t, 403, status)
+	jsonAssert(t, body, map[string]any{
+		"error": "You do not own this coin",
```

Comment:
I continue to think that testing the exact content of these messages is brittle and unnecessary. But I might be in the minority. :-)